src/super_gradients/common/pro_user/monitoring.py

```diff
@@ -0,0 +1,92 @@
+    _imported_deci_lab_failure = _import_err
+
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+
+class ExceptionInfo:
+    """Holds information about the session exception (if any)"""
+
+    _is_exception_raised = False
+
+    @staticmethod
+    def register_exception():
+        """Register the exception information into the class"""
+        ExceptionInfo._is_exception_raised = True
+
+    @staticmethod
+    def is_exception_raised():
+        """Check if an exception was raised in the current process"""
+        return ExceptionInfo._is_exception_raised
+
+
+def register_exceptions(excepthook: Callable):
+    """Wrap excepthook with a step the saves the exception info to be available in the exit hooks."""
+
+    def excepthook_with_register(exc_type, exc_value, exc_traceback):
+        ExceptionInfo.register_exception()
+        return excepthook(exc_type, exc_value, exc_traceback)
+
+    return excepthook_with_register
+
+
+@multi_process_safe
+def exception_upload_handler(platform_client):
+    """Upload the log file to the deci platform if an error was raised"""
+    # Make sure that the sink is flushed
+    ConsoleSink.flush()
+    if not is_distributed() and ExceptionInfo.is_exception_raised():
+
+        logger.info("Uploading console log to deci platform ...")
+
+        data = platform_client.upload_log_url(tag="SuperGradients", level="error")
+        signed_url = S3SignedUrl(**data.data)
+        platform_client.upload_file_to_s3(from_path=ConsoleSink.get_filename(), s3_signed_url=signed_url)
+
+        # TODO: multiprocess safe
+        # time.sleep(10)
+        logger.info("Exception was uploaded to deci platform!")
+
+
+def setup_pro_user_monitoring():
+    """Setup the pro user environment for error logging and monitoring"""
+    upload_console_logs = os.getenv("UPLOAD_LOGS", "TRUE") == "TRUE"
+    if upload_console_logs:
+        logger.info("You installed pro user libraries, so we assume that you have a pro user account.")
+        logger.info("If you are not a pro user, or if you don't want to connect to the deci platform, set the env variable UPLOAD_LOGS=FALSE")
+
+        # This prevents hydra.main to catch errors that happen in the decorated function
+        # (which leads sys.excepthook to never be called)
+        os.environ["HYDRA_FULL_ERROR"] = "1"
+
+        logger.info("Connecting to the deci platform ...")
+        platform_client = DeciPlatformClient()
+        platform_client.login(token=os.getenv("DECI_PLATFORM_TOKEN"))
+        logger.info("Connection to the deci platform successful!")
+
+        sys.excepthook = register_exceptions(sys.excepthook)
+        atexit.register(exception_upload_handler, platform_client)
+    else:
+        logger.info("You disabled the upload of your logs to deci platform." "You can reactivate it by setting the env variable UPLOAD_LOGS=TRUE")
+
+
+def setup_user_env():
+    if _imported_deci_lab_failure is None:
+        setup_pro_user_monitoring()
```

Comment:
This means that if someone has the pro requirements we will by default try to upload to platform, unless an env variable is set.
This would apply to most deci members, so I guess everyone will need to set the env variable to skip it.
Are we fine with that ? Isnt there another way ?

Reply:
can't we check if the user is logged in?

Reply:
There is a private function that we could use, and we could also ask platform team to make it public.
The main problem I see is that in that case if a pro user is not logged in yet (or deleted his token for any reason) then this feature would be disabled by default, and I guess we would rather avoid this situation.